datafusion/sql/src/statement.rs

```diff
@@ -681,8 +680,22 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             order_exprs,
             unbounded,
             options,
+            constraints,
         } = statement;
 
+        let mut constraints = constraints;
+        for column in &columns {
+            for option in &column.options {
+                if let ast::ColumnOption::Unique { is_primary } = option.option {
+                    constraints.push(ast::TableConstraint::Unique {
+                        name: None,
+                        columns: vec![column.name.clone()],
+                        is_primary,
+                    })
+                }
+            }
```

Comment:
My reading of this code is that other types of constraints will be silently ignored. Perhaps we can error if we see an unknown constraint?

In fact, I think if this code called

```
        let constraints =
            Constraints::new_from_table_constraints(&constraints, &df_schema)?;
```

It would return errors for unsupported constraints already